src/fsharp/opt.fs

```diff
@@ -2139,17 +2137,8 @@ and OptimizeTryFinally cenv env (spTry,spFinally,e1,e2,m,ty) =
           HasEffect = e1info.HasEffect || e2info.HasEffect;
           MightMakeCriticalTailcall = false; // no tailcalls from inside in try/finally
           Info = UnknownValue } 
-    (* try-finally, so no effect means no exception can be raised, so just sequence the finally *)
-    if cenv.settings.EliminateTryCatchAndTryFinally () && not e1info.HasEffect then 
```

Comment:
We could just set the flag "EliminateTryCatchAndTryFinally" to false.  (The optimization is valid in some situations - over an IL "add" integer instruction for example).  But I don't mind too much either way, it's your call.